server/plugin/configuration.go

```diff
@@ -144,9 +144,10 @@ func (p *Plugin) OnConfigurationChange() error {
 	p.setConfiguration(configuration)
 
 	if oldEncryptionSecret != "" && oldEncryptionSecret != p.getConfiguration().EncryptionSecret {
-		if err := p.store.DeleteUserTokenOnEncryptionSecretChange(); err != nil {
-			p.API.LogError("Error in deleting Users.", "Error", err.Error())
-			return err
+		ch := make(chan error, 1)
+		go p.store.DeleteUserTokenOnEncryptionSecretChange(ch)
+		if len(ch) == 1 {
+			p.API.LogError("Error in deleting Users.", "Error", (<-ch).Error())
```

Comment:
By doing this, you are waiting for the goroutine to execute and add values in the channel but we don't have to wait for it.